src/formatting.rs

```diff
@@ -39,20 +30,51 @@ fn unexpected_token<T>(token: char) -> Result<T> {
     ))
 }
 
+#[derive(Debug, Clone)]
+pub struct FormatTemplate {
+    full: Option<Vec<Token>>,
+    short: Option<Vec<Token>>,
+}
+
 impl FormatTemplate {
     /// Whether the format string contains given placeholder
     pub fn contains(&self, var: &str) -> bool {
-        for token in &self.tokens {
-            if let Token::Var(ref placeholder) = token {
-                if placeholder.name == var {
-                    return true;
```

Comment:
I think `FormatTemplate::contains()` should take `short` into account too.

Reply:
As in check both? I figured everything in `short` would also be in `full` but i guess that's reasonable